### PR TITLE
Update chart loading

### PR DESCRIPTION
**Pull request by kalletlak**

### CODE REVIEW

File: src/pages/studyView/charts/StudyViewComponentLoader.tsx [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
-                <If condition={this.props.promise.isPending}>
-                    <LoadingIndicator
-                        isLoading={!!this.props.promise.isPending}
-                    />
-                </If>
-                <If condition={this.props.promise.isError}>
-                    <div>Error when loading data.</div>
-                </If>
-                <If condition={this.props.promise.isComplete}>
+            <div style={{ position: "relative", height: "100%"}}>
```
Comment (zhx828) [inline src/pages/studyView/charts/StudyViewComponentLoader.tsx]: Why don't we use LoadingIndicator? And this style is lost `styles.studyViewAutoMargin`

Comment (kalletlak) [reply src/pages/studyView/charts/StudyViewComponentLoader.tsx]: I wasn't able to apply styles to that loader

Comment (alisman) [reply src/pages/studyView/charts/StudyViewComponentLoader.tsx]: LoadingIndicator should have a style property which will apply styles directly to loader.  Please do use it.  Check definition.

File: src/pages/studyView/charts/ChartContainer.tsx [hunk 2]
```diff
@@ -258,7 +267,7 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
                     chartControls={this.chartControls}
                     changeChartType={this.changeChartType}
                 />
-                <StudyViewComponentLoader promise={this.props.promise}>
+                <StudyViewComponentLoader promise={this.props.promise} isInitialLoaded={!_.isUndefined(this.chartData)}>
```
Comment (alisman) [inline src/pages/studyView/charts/ChartContainer.tsx]: You can now just use === undefined.     The problem with testing against undefined is old and has been solved in modern browsers.

File: src/pages/studyView/charts/ChartContainer.tsx [hunk 3]
```diff
@@ -154,6 +156,13 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
         this.chartType = chartType;
     }
 
+    @computed get chartData(){
```
Comment (alisman) [inline src/pages/studyView/charts/ChartContainer.tsx]: how come we are caching data?

**Comment by kalletlak** [comment]
Closing this as we no longer require this functionality

**Pull request closed without merging**